scripts/superdesk/editor2/editor.js

```diff
@@ -325,19 +325,26 @@ function EditorService(spellcheck, $rootScope, $timeout, $q, _, renditionsServic
      * @param {Boolean} preventStore
      */
     function hilite(node, tokens, className, preventStore) {
-        if (!tokens.length) {
-            self.resetSelection(node);
-            return;
-        }
+        var CLONE_CLASS = 'clone';
+        var parentNode = node.parentNode;
 
-        if (!preventStore) {
-            self.storeSelection(node);
-        }
-        var token = tokens.shift();
-        hiliteToken(node, token, className);
-        $timeout(function() {
-            hilite(node, tokens, className, true);
-        }, 0, false);
+        // remove old hilite nodes if any
+        var clones = parentNode.getElementsByClassName(CLONE_CLASS);
+        Array.prototype.forEach.call(clones, function(clone) {
+            parentNode.removeChild(clone);
+        });
+
```

Comment:
getElementsByClassName returns a live collection, would be better to use slice to make an actual array from this DOM  collection first.